test/unit/config/index.js

```diff
@@ -0,0 +1,291 @@
+            let oldArgv;
+            let oldEnv;
+
+            beforeEach(() => {
+                oldArgv = [].concat(process.argv);
+                oldEnv = _.clone(process.env);
+            });
+
+            afterEach(() => {
+                process.env = oldEnv;
+                process.argv = oldArgv;
+            });
+
+            it('should override option in config with CLI argument beginning with `--`', () => {
+                process.argv = process.argv.concat('--username', 'argv_username');
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'config_username',
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('argv_username');
+            });
+
+            it('should override option in config with ENV variable prefixed with `docpub_`', () => {
+                process.env['docpub_username'] = 'env_username';
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'config_username',
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('env_username');
+            });
+
+            it('should use argv option, if both cli and argv options available', () => {
+                process.env['docpub_username'] = 'env_username';
+                process.argv = process.argv.concat('--username', 'argv_username');
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'config_username',
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('argv_username');
+            });
+
+            it('should allow to not declare options in json config if argv option available', () => {
+                process.argv = process.argv.concat('--username', 'argv_username');
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('argv_username');
+            });
+
+            it('should allow to not declare options in json config if env option available', () => {
```

Comment:
`should allow options to not be declared in json config if env option is available`?